index.d.ts

```diff
@@ -0,0 +1,73 @@
+import { type EventEmitter } from 'events'
+import type bunyan from 'bunyan'
+
+export as namespace hato;
+
+/**
+ * Plugins
+ */
+export class Plugin {
+    constructor(name?: string)
+}
+
+export class RPC extends Plugin {
+    constructor({ timeout }: { timeout: number })
+}
+
+export class Retry extends Plugin {
+    constructor({ base }: { base: number })
+}
+
+export class ServiceContext extends Plugin {
+    constructor({ name, version }: { name: string, version:  string })
+}
+
+type Defaults = {
+    prefetch: number
+    queue: {
+        durable: boolean
+    },
+    exchange: {
+        durable: boolean
+    }
+    publish: {
+        persistent: boolean
+        mandatory: boolean
+    }
+}
+
+export class DefaultOptions extends Plugin {
+    constructor(options: Defaults)
+}
+
+export const plugins: {
+    GracefulShutdown: typeof Plugin,
+    ConnectionRetry: typeof Plugin,
+    Duplex: typeof Plugin,
+    DLX: typeof Plugin,
+    Encoding: typeof Plugin,
+    RPC: typeof RPC,
+    Confirm: typeof Plugin,
+    Retry: typeof Retry,
+    ServiceContext: typeof ServiceContext,
+    DefaultOptions: typeof DefaultOptions,
+    Base: typeof Plugin
+}
+
+/**
+ * Client
+ */
+
+type Options = {
+    logger: bunyan
```

Comment:
I think we can declare an interface with `debug` `info` `warn` `error` that is `(s: string) => void` or `(v: any) => void` (whichever works) and not include bunyan in the dependency

Reply:
Sure, I did this.